crates/circuit/src/circuit_instruction.rs

```diff
@@ -878,24 +878,49 @@ pub fn convert_py_to_operation_type(
         Ok(stdgate) => stdgate.extract().ok().unwrap_or_default(),
         Err(_) => None,
     };
-    // If the input instruction is a standard gate and a singleton instance
+    // If the input instruction is a standard gate and a singleton instance,
     // we should check for mutable state. A mutable instance should be treated
     // as a custom gate not a standard gate because it has custom properties.
-    //
-    // In the futuer we can revisit this when we've dropped `duration`, `unit`,
+    // Controlled gates with non-default control states are also considered
+    // custom gates even if a standard representation exists for the default
+    // control state.
+
+    // In the future we can revisit this when we've dropped `duration`, `unit`,
     // and `condition` from the api as we should own the label in the
     // `CircuitInstruction`. The other piece here is for controlled gates there
     // is the control state, so for `SingletonControlledGates` we'll still need
     // this check.
     if standard.is_some() {
         let mutable: bool = py_op.getattr(py, intern!(py, "mutable"))?.extract(py)?;
-        if mutable
+        // The default ctrl_states are the all 1 state and None.
+        // These are the only cases where controlled gates can be standard.
+        let is_default_ctrl_state = || match py_op.getattr(py, intern!(py, "ctrl_state")) {
+            Ok(c_state) => match c_state.extract::<Option<i32>>(py) {
+                Ok(c_state_int) => match c_state_int {
+                    Some(c_int) => {
+                        let qubits: u32 = py_op
+                            .getattr(py, intern!(py, "num_qubits"))
+                            .unwrap()
+                            .extract(py)
+                            .unwrap();
```

Comment:
I think we can preserve the errors here with:

```suggestion
                            .getattr(py, intern!(py, "num_qubits"))?
                            .extract(py)?;
```

Then in L918 we can just do `!is_default_ctrl_state()?)` to return that back to the users.

Reply:
This was my original idea, but adding the `?` triggers this annoying compilation errror:

```
error[E0277]: the `?` operator can only be used in a closure that returns `Result` or `Option` (or another type that implements `FromResidual`)
   --> crates/circuit/src/circuit_instruction.rs:902:68
    |
897 |         let is_default_ctrl_state = || match py_op.getattr(py, intern!(py, "ctrl_state")) {
    |                                     -- this function should return `Result` or `Option` to accept `?`
...
902 |                             .getattr(py, intern!(py, "num_qubits"))?
    |                                                                    ^ cannot use the `?` operator in a closure that returns `bool`
    |
    = help: the trait `FromResidual<Result<Infallible, pyo3::PyErr>>` is not implemented for `bool`
```

Reply:
Ok, either I'm missing something or it's not so easy to preserve the error. Changing the return type doesn't help: 
```
the trait `FromResidual<Result<Infallible, pyo3::PyErr>>` is not implemented for `Option<bool>`
```

Reply:
No you're not missing something obvious. This is a common problem. You can find buzz about it online. I finally found a clean solution for a PR yesterday. I'll look it up and see if it applies here.

I don't think the stuff I found online was directly relevant to the solution I found. In fact I think a lot of the discussion was not helpful. You can search for variants of : "rust question mark operator return  closure"

Reply:
Here is a PR to this branch https://github.com/ElePT/qiskit/pull/34 that appears return `PyResult` rather than panic on the two `unwrap`s.

Reply:
Thanks a lot @jlapeyre, it would have taken me a long time to find the fix. I will keep it in mind for future PRs.